### PR TITLE
Add types for `String#isWellFormed` and `String#toWellFormed`

### LINKED ISSUE #55543: Support `isWellFormed` and `toWellFormed` in lib.esnext.d.ts
### 🔍 Search Terms
NodeJS 20, isWellFormed, toWellFormed

### ✅ Viability Checklist

- [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
- [X] This wouldn't change the runtime behavior of existing JavaScript code
- [X] This could be implemented without emitting different JS based on the types of the expressions
- [X] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
- [X] This feature would agree with the rest of our Design Goals: https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals

### ⭐ Suggestion

NodeJS 20 introduces new features that are not supported by the TypeScript compiler yet.

As an example node version 20 introduces `String.prototype.isWellFormed` and `String.prototype.toWellFormed`. Using these functions causes `error TS2339: Property 'isWellFormed' does not exist on type 'string'.`

Current environment:
- Node version: 20.5.1
- TypeScript version: 5.2.2

### 📃 Motivating Example

This update will enable developers to fully use all features available in the latest major version of NodeJS.

### 💻 Use Cases

This would allow the usage of ECMAScript features such as `isWellFormed` and `toWellFormed`.

The only current workaround is to use `// @ts-ignore` to let the project compile.

### PR DESCRIPTION
**Pull request by andreubotella**
These two methods are part of the [well-formed Unicode strings](https://github.com/tc39/proposal-is-usv-string) proposal, which reached stage 4 in May 2023 and will be part of ES2024.

Fixes #55543.

### CODE REVIEW

**Comment by andreubotella** [comment]
@microsoft-github-policy-service agree

File: src/lib/esnext.string.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+interface String {
+    /**
+     * Returns true if the string does not contain any lone UTF-16 surrogates.
```
Comment (sandersn) [inline src/lib/esnext.string.d.ts:3]: I'd rather use the description in the proposal:

```suggestion
     * Returns true if all leading surrogates and trailing surrogates appear paired and in order.
```

File: src/lib/esnext.string.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+interface String {
+    /**
+     * Returns true if the string does not contain any lone UTF-16 surrogates.
+     */
+    isWellFormed(): boolean;
+
+    /**
+     * Returns a string where all lone UTF-16 surrogates of this string have been replaced by the Unicode replacement character (U+FFFD).
```
Comment (sandersn) [inline src/lib/esnext.string.d.ts:8]: Same here: 

```suggestion
     * Returns a string where all lone or out-of-order UTF-16 surrogates have been replaced by the Unicode replacement character (U+FFFD).
```

I'm not sure whether UTF-16 needs to be mentioned or not.

**Review by sandersn — CHANGES_REQUESTED** [review]
Structure looks good. I'd like to stay closer to the method descriptions in the proposal, though.

**Review by andrewbranch — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks for the PR! To help users out a bit: in `utilities.ts` there's a function called `getScriptTargetFeatures` that recommends using the appropriate `lib`/`target` setting when someone accesses values that don't seem to exist. Can you update that?

**Comment by andreubotella** [comment]
> Thanks for the PR! To help users out a bit: in `utilities.ts` there's a function called `getScriptTargetFeatures` that recommends using the appropriate `lib`/`target` setting when someone accesses values that don't seem to exist. Can you update that?

Done

**Comment by DanielRosenwasser** [comment]
@sandersn are we good to merge?

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**